src/main/java/uk/nhs/tis/trainee/notifications/service/EmailService.java

```diff
@@ -120,8 +121,89 @@ public void sendMessageToExistingUser(String traineeId, NotificationType notific
   public void sendMessage(String traineeId, String recipient, NotificationType notificationType,
       String templateVersion, Map<String, Object> templateVariables,
       TisReferenceInfo tisReferenceInfo, boolean doNotSendJustLog) throws MessagingException {
+
     String templateName = templateService.getTemplatePath(EMAIL, notificationType, templateVersion);
     log.info("Sending template {} to {}.", templateName, recipient);
+    ObjectId notificationId = ObjectId.get();
+
+    MimeMessageHelper helper = buildMessageHelper(recipient, templateName, templateVariables,
+        notificationId);
+
+    if (!doNotSendJustLog) {
+      mailSender.send(helper.getMimeMessage());
+
+      // Store the notification history.
+      RecipientInfo recipientInfo = new RecipientInfo(traineeId, EMAIL, recipient);
+      TemplateInfo templateInfo = new TemplateInfo(notificationType.getTemplateName(),
+          templateVersion, templateVariables);
+      History history = new History(notificationId, tisReferenceInfo, notificationType,
+          recipientInfo, templateInfo, Instant.now(), null, NotificationStatus.SENT, null, null);
+      historyService.save(history);
+
+      log.info("Sent template {} to {}.", templateName, recipient);
+    } else {
+      try {
+        log.info("For now, just logging mail to '{}' with subject '{}' and content '{}'", recipient,
+            helper.getMimeMessage().getSubject(), helper.getMimeMessage().getContent().toString());
+      } catch (IOException e) {
+        //should not happen
+        log.info("For now, just logging mail to '{}' with subject '{}' and unparseable content",
```

Comment:
I don't know how to test this unfortunately 😖

Reply:
You can verify that the IOException is swallowed and doesn't trigger any other type of exception

```java
@Test
void coverage() throws IOException, MessagingException {
  MimeMessage mimeMessage = mock(MimeMessage.class);
  when(mailSender.createMimeMessage()).thenReturn(mimeMessage);

  when(mimeMessage.getContent()).thenThrow(IOException.class);

  assertDoesNotThrow(() -> service.sendMessage(TRAINEE_ID, RECIPIENT, NOTIFICATION_TYPE, "v1.2.3",
      new HashMap<>(), null, true));
}
```

Reply:
Ah, mocking the MimeMessage, that I didn't think of! 🌮